mewarpx/medocker/docker_shared/run_warpx.sh

```diff
@@ -119,21 +156,30 @@ do
     find . -type f -exec touch -a -c {} +
 done
 }
+
 # https://stackoverflow.com/questions/21465297/tee-stdout-and-stderr-to-separate-files-while-retaining-them-on-their-respective
-{ { bash $SCRIPTNAME; } > >( tee -a stdout.out ); } \
+bash $SCRIPTNAME > >( tee -a stdout.out ) \
                        2> >( tee -a stderr.err >&2 ) &
 
+# sleep so that there is time for the mpiexec process to launch
+sleep 2
+
 MAINID=$!
+MPIEXECID=$(pgrep -P $MAINID -f mpiexec)
+echo "main process id is $MAINID"
+echo "mpiexec process id: $MPIEXECID"
 checkpoint &
 CHECKID=$!
-wait -n $MAINID $CHECKID
+pingterminatewarning &
+PINGID=$!
+wait $MAINID
```

Comment:
```suggestion
wait -n $MAINID $CHECKID
```

I think this is still correct/necessary. `-n` means it proceeds if either exits. Normally it's `$MAINID` that exits. `checkpoint` exits if it times out, however. So I'm pretty sure with only `wait $MAINID` that timeout will kill `checkpoint` but not the full script, which is not desired behavior (eg WarpX will run indefinitely if it freezes for some reason).